### PR TITLE
2.x: Async redis

### PR DESCRIPTION
**Pull request by adam-fowler**
Creates new type HBRedisConnectionPoolService which conforms to Service
- Uses only one RedisConnectionPool (no point have multiple pools aligned with EventLoops anymore)
- Updated to included failed queue

### CODE REVIEW

File: Sources/HummingbirdJobsRedis/RedisJobQueue.swift [hunk 1]
```diff
@@ -34,155 +37,167 @@ public final class HBRedisJobQueue: HBJobQueue {
...
     }
 
-    func set(jobId: JobIdentifier, job: HBJobContainer, pool: RedisConnectionPool) -> EventLoopFuture<Void> {
-        return pool.set(jobId.redisKey, toJSON: job)
+    /// Push all the entries from list back onto the main list.
+    func rerun(queueKey: RedisKey) async throws {
+        while true {
+            let key = try await self.redisConnectionPool.rpoplpush(from: queueKey, to: self.configuration.queueKey).get()
+            if key.isNull {
+                break
```
Comment (Joannis) [inline Sources/HummingbirdJobsRedis/RedisJobQueue.swift:139]: Can we make this `return` instead?

File: Tests/HummingbirdRedisTests/PersistTests.swift [hunk 2]
```diff
@@ -14,229 +14,426 @@
...
+                XCTAssertEqual(response.status, .ok)
+            }
+            try await client.XCTExecute(uri: "/persist/\(tag)", method: .get) { response in
+                XCTAssertEqual(response.status, .ok)
+                let body = try XCTUnwrap(response.body)
+                XCTAssertEqual(String(buffer: body), "ThisIsTest1")
+            }
         }
     }
+    /*        let app = HBApplication(testing: .live)
```
Comment (Joannis) [inline Tests/HummingbirdRedisTests/PersistTests.swift:220]: Lots of commented tests still

File: Sources/HummingbirdJobsRedis/RedisJobQueue.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/HummingbirdJobsRedis/RedisJobQueue.swift]: fixed

File: Tests/HummingbirdRedisTests/PersistTests.swift (hunk 2 above)
Comment (adam-fowler) [reply Tests/HummingbirdRedisTests/PersistTests.swift]: Removed. All relevant commented tests have equivalent versions implemented

**Pull request merged**